spring-cloud-function-adapters/spring-cloud-function-adapter-google-cloud-function/src/main/java/org/springframework/cloud/function/adapter/gcloud/GcloudSpringBootHttpRequestHandler.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.function.adapter.gcloud;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.cloud.functions.Context;
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.gson.Gson;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+
+import org.springframework.cloud.function.context.AbstractSpringFunctionAdapterInitializer;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.support.GenericMessage;
+
+/**
+ * @author Dmitry Solomakha
+ */
+public class GcloudSpringBootHttpRequestHandler<O>
+	extends AbstractSpringFunctionAdapterInitializer<Context> implements HttpFunction {
+
+	Gson gson = new Gson();
+
+	public GcloudSpringBootHttpRequestHandler() {
+		super();
+	}
+
+	public GcloudSpringBootHttpRequestHandler(Class<?> configurationClass) {
+		super(configurationClass);
+	}
+
+	protected Object convert(HttpRequest event) throws IOException {
+		BufferedReader br = event.getReader();
+		StringBuilder sb = new StringBuilder();
+		String line;
+		while ((line = br.readLine()) != null) {
```

Comment:
Might be better to read into a `char[]` and append to stringbuilder to preserve the true line endings since different platforms may not always use `\n`